appveyor.yml

```diff
@@ -1,71 +1,84 @@
 version: '{build}'
 
-os: Visual Studio 2015
-
 environment:
   matrix:
-  - Toolset: v140
-  - Toolset: v120
-  - Toolset: v110
-  - Toolset: v100
+    - compiler: msvc-10-seh
+      generator: "Visual Studio 10 2010"
+
+    - compiler: msvc-11-seh
+      generator: "Visual Studio 11 2012"
+
+    - compiler: msvc-11-seh
+      generator: "Visual Studio 11 2012 Win64"
+
+    - compiler: msvc-12-seh
+      generator: "Visual Studio 12 2013"
+
+    - compiler: msvc-12-seh
+      generator: "Visual Studio 12 2013 Win64"
 
-platform:
-  - Win32
-  - x64
+    - compiler: msvc-14-seh
+      generator: "Visual Studio 14 2015"
+
+    - compiler: msvc-14-seh
+      generator: "Visual Studio 14 2015 Win64"
+
+    - compiler: gcc-5.3.0-posix
+      generator: "MinGW Makefiles"
 
 configuration:
-#  - Release
   - Debug
+  - Release
 
 build:
   verbosity: minimal
 
-artifacts:
-  - path: '_build/Testing/Temporary/*'
-    name: test_results
-
-before_build:
+install:
 - ps: |
-    Write-Output "Configuration: $env:CONFIGURATION"
-    Write-Output "Platform: $env:PLATFORM"
-    $generator = switch ($env:TOOLSET)
-    {
-        "v140" {"Visual Studio 14 2015"}
-        "v120" {"Visual Studio 12 2013"}
-        "v110" {"Visual Studio 11 2012"}
-        "v100" {"Visual Studio 10 2010"}
-    }
-    if ($env:PLATFORM -eq "x64")
-    {
-        $generator = "$generator Win64"
+    Write-Output "Compiler: $env:compiler"
+    Write-Output "Generator: $env:generator"
+
+    # git bash conflicts with MinGW makefiles
+    if ($env:generator -eq "MinGW Makefiles") {
+        $env:path = $env:path.replace("C:\Program Files\Git\usr\bin;", "")
+        if ($env:compiler -eq "gcc-5.3.0-posix") {
+            $cxx_path = "C:\mingw-w64\i686-5.3.0-posix-dwarf-rt_v4-rev0\mingw32\bin"
+        }
+        if ($cxx_path -ne "") {
+            $env:path += ";$cxx_path"
+        }
     }
 
 build_script:
 - ps: |
-    if (($env:TOOLSET -eq "v100") -and ($env:PLATFORM -eq "x64"))
-    {
-        return
-    }
     md _build -Force | Out-Null
     cd _build
 
-    & cmake -G "$generator" -DCMAKE_CONFIGURATION_TYPES="Debug;Release" -Dgtest_build_tests=ON -Dgtest_build_samples=ON -Dgmock_build_tests=ON ..
+    $conf = if ($env:generator -eq "MinGW Makefiles") {"-DCMAKE_BUILD_TYPE=$env:configuration"} else {"-DCMAKE_CONFIGURATION_TYPES=Debug;Release"}
+    # Disable test for MinGW (gtest tests fail, gmock tests can not build)
```

Comment:
Currently, builds are failing anyway, so the maintainers probably merging after manual quick-look.
So, if mingw builds fail, the maintainers may assume that it is not the PR author's fault. I don't know what the maintainers want, though. It would be helpful to get their opinion.